File: java/src/br/unifor/facilitador/infraestrutura/NotificadorEmail.java
package br.unifor.facilitador.infraestrutura;

import br.unifor.facilitador.dominio.Mensagem;
import br.unifor.facilitador.dominio.Usuario;
import br.unifor.facilitador.porta.Notificador;

public class NotificadorEmail implements Notificador {

    private final String servidorSmtp;

    public NotificadorEmail(String servidorSmtp) {
        this.servidorSmtp = servidorSmtp;
    }

    @Override
    public void notificar(Usuario u, Mensagem msg) {
        System.out.println("[email via " + servidorSmtp + "] para " + u.getEmail()
                + " | " + msg.getTitulo() + ": " + msg.getCorpo());
    }
}

File: src/NotificadorEmail.ts
import Notificador from "./Notificador.js";
import Usuario from "./Usuario.js";
import Mensagem from "./Mensagem.js";

export default class NotificadorEmail implements Notificador {
  servidorSmtp: string;

  notificar(u: Usuario, msg: Mensagem): void {}
}
